### PR TITLE
fix concurrency issue in H2CommTask

### PR DESCRIPTION
**Pull request by graetzer**
make h2spec work again

### CODE REVIEW

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/10842/

File: arangod/GeneralServer/H2CommTask.cpp [hunk 1]
```diff
@@ -154,7 +161,10 @@ template <SocketType T>
   H2CommTask<T>* me = static_cast<H2CommTask<T>*>(user_data);
   auto it = me->_streams.find(stream_id);
   if (it != me->_streams.end()) {
-    it->second->statistics.SET_WRITE_END();
+    Stream& strm = *it->second;
+    if (strm.response && strm.response->statistics) {
```
Comment (mpoeter) [inline arangod/GeneralServer/H2CommTask.cpp:165]:
```suggestion
    if (strm.response) {
```
No need to check `statistics`, this is implicitly done in `SET_WRITE_END()` & co.

File: arangod/GeneralServer/H2CommTask.cpp [hunk 2]
```diff
@@ -656,7 +660,10 @@ void H2CommTask<T>::queueHttp2Responses() {
       };
       prd_ptr = &prd;
     }
-    stat.ADD_SENT_BYTES(res.bodySize());
+    
+    if (res.statistics) {
+      res.statistics.ADD_SENT_BYTES(res.bodySize());
+    }
```
Comment (mpoeter) [inline arangod/GeneralServer/H2CommTask.cpp:666]:
```suggestion
    res.statistics.ADD_SENT_BYTES(res.bodySize());
```

**Comment by jsteemann** [comment]
https://172.16.10.101/job/arangodb-matrix-pr/10849/

**Comment by jsteemann** [comment]
https://172.16.10.101/job/arangodb-matrix-pr/10850/

**Pull request merged**